data/rao-result/rao-result-json/src/main/java/com/powsybl/openrao/data/raoresultjson/serializers/FlowCnecResultArraySerializer.java

```diff
@@ -121,7 +124,7 @@ private static void serializeFlowCnecFlows(Instant optInstant, Unit unit, FlowCn
 
         jsonGenerator.writeObjectFieldStart(serializeSide(side));
         if (!Double.isNaN(flow)) {
-            jsonGenerator.writeNumberField(FLOW, Math.round(100.0 * flow) / 100.0);
+            jsonGenerator.writeNumberField(FLOW, roundDoubleValue(flow, Math.max(2, computeNumberOfRelevantDecimals(margin))));
```

Comment:
because that was requested for this new feature, the number of decimals to display for a measurement must be increased only if the margin is very narrow and would become "invisible" in the results because of the rounding